src/SimpleRewriter.php

```diff
@@ -10,50 +10,55 @@
 
 class SimpleRewriter {
 
-    /**
-     * SimpleRewriter constructor
-     */
-    public function __construct() {
-
-    }
-
     /**
      * Rewrite URLs in file to destination_url
      *
      * @param string $filename file to rewrite URLs in
      * @throws WP2StaticException
      */
     public static function rewrite( string $filename ) : void {
-        $destination_url = apply_filters(
-            'wp2static_set_destination_url',
-            CoreOptions::getValue( 'deploymentURL' )
-        );
+        $file_contents = file_get_contents( $filename );
 
-        $wordpress_site_url =
-            apply_filters(
-                'wp2static_set_wordpress_site_url',
-                untrailingslashit( SiteInfo::getUrl( 'site' ) )
-            );
+        if ( $file_contents === false ) {
+            $file_contents = '';
+        }
 
-        $file_contents = file_get_contents( $filename );
+        $rewritten_contents = self::rewriteFileContents( $file_contents );
+
+        file_put_contents( $filename, $rewritten_contents );
+    }
 
+    /**
+     * Rewrite URLs in a string to destination_url
+     *
+     * @param string $file_contents
+     * @return string
+     */
+    public static function rewriteFileContents( string $file_contents ) : string
+    {
         // TODO: allow empty file saving here? Exception for style.css
         if ( ! $file_contents ) {
-            return;
+            return '';
         }
 
+        $destination_url = apply_filters(
+            'wp2static_set_destination_url',
+            CoreOptions::getValue( 'deploymentURL' )
+        );
+
+        $wordpress_site_url = apply_filters(
+            'wp2static_set_wordpress_site_url',
+            trailingslashit( SiteInfo::getUrl( 'site' ) )
+        );
+
         $search_patterns = [
-            trailingslashit( $wordpress_site_url ),
-            $wordpress_site_url,
             addcslashes( $wordpress_site_url, '/' ),
             addcslashes( trailingslashit( $wordpress_site_url ), '/' ),
```

Comment:
What are these for? Is just replacing the protocol relative site URL with destination URL not enough?

Reply:
I believe this was to cover say:

original WP: http://www.mydevsite.com

destination: https://nakeddomain.com

When these appear cslashed in inline JS or such:

http\:\/\/www\.mydevsite\.com (not 100% remembering if all those chars are slashed, but you get the idea)

Reply:
Ahh, this makes a lot of sense. Thanks!